application-example.conf

```diff
@@ -89,45 +137,35 @@ kafka {
 
   maxRequestSize: 250000000
 
-}
+  securityProtocol: "SSL"
 
-worker {
+  consumerPropertyFile: ".../consumer-conf/consumer.properties"
 
-  # maximum number of times across all workers that an attempt should be made to process any given document;
-  # when this property is not provided, retries will not be tracked and no limit will be imposed;
-  # this property requires that zookeeper is available at the specified zookeeper.connectString
-  #maxRetries: 2
+  producerPropertyFile: ".../consumer-conf/producer.properties"
 
-  # number of worker threads to start for each pipeline when running lucille in local / single-JVM mode
-  threads: 2
+  documentDeserializer: # TODO
 
-  # maximum time to spend processing a message, after which the worker will assume a problem and shut down
-  # NOTE: not supported while com.kmwllc.lucille.core.Worker and PipelineWorker are being reconciled
-  #maxProcessingSecs: 600 # 10 minutes
-}
+  documentSerializer: # TODO
 
-publisher {
+  events: # TODO
 
-  # this setting controls
-  #     1) maximum size of the queue of published documents waiting to be processed
-  #     2) maximum size of the queue of completed documents waiting to be indexed
-  # e.g. if it is set to 10 then each each queue can contain up to 10 documents for a total of 20
-  #
-  # attempts to publish beyond this capacity will cause the publisher to block
-  # this setting applies only when running Lucille in local / single-JVM mode using a LocalMessageManager
-  # this setting defaults to 100
-  queueCapacity: 100
+  sourceTopic: # TODO
```

Comment:
this is the name of the kafka topic that contains Documents generated by the Publisher that are waiting to be picked up by a Worker and sent through the pipeline. Typically, lucille generates this topic "behind the scenes" and names it as <pipeline_name>_source
However, there are cases where Lucille runs without a Runner/Publisher and doesn't create the sourceTopic; rather an existing Kafka topic should be used as sourceTopic and a lucille Worker or WorkerIndexer should read Documents from that existing topic. In that scenario, a user would want to specify kafka.sourceTopic in the config and make sure it's set to whatever topic contains the documents that they want to process

Reply:
Thanks! Addressed in new commit